crypto/evp/digest.c

```diff
@@ -174,6 +174,27 @@ int EVP_DigestFinal_ex(EVP_MD_CTX *ctx, unsigned char *md, unsigned int *size)
     return ret;
 }
 
+int EVP_DigestFinalXOF(EVP_MD_CTX *ctx, unsigned char *md, size_t size)
+{
+    int ret;
+
+    if (ctx->digest->flags & EVP_MD_FLAG_XOF && size <= INT_MAX &&
+        ctx->digest->md_ctrl(ctx, EVP_MD_CTRL_XOF_LEN, (int)size, NULL)) {
+        ret = ctx->digest->final(ctx, md);
+
+        if (ctx->digest->cleanup != NULL) {
+            ctx->digest->cleanup(ctx);
+            EVP_MD_CTX_set_flags(ctx, EVP_MD_CTX_FLAG_CLEANED);
+        }
+        OPENSSL_cleanse(ctx->md_data, ctx->digest->ctx_size);
+
+        return ret;
+    } else {
```

Comment:
Note that style guide doesn't dictate use of 'goto err': "The goto statement comes in handy when a function exits from multiple locations." In other words goto is [customarily] considered as sign of desperation, but you do what you have to do, but only *when* you have to do it. This is not case here, so I'd argue against goto. Function is short and simple, no multiple exits, no need for 'goto'. Now, one can argue in favour of omission of else. I wouldn't add it if not for EVPerr. I find it appropriate to explicitly denote that it's outcome of failed condition. I mean without 'else' UNSPECIFIED_XOF_ERROR would probably be more appropriate with EVPerr. Wouldn't you agree?